techmatters/terraso-mobile-client
# feat: select soil

## Description

Add the ability to select a soil ID match from the Soil Info sheet. The selected match will display above the list of matches in the soil ID screen and will take precedence over the top match for the dashboard screen's location soil ID info card. The selection is stored in a new "Soil Metadata" model. Selection read/write is exposed through a hook.

Also refactor some nearby code for ease of making changes.

### Checklist

### Related Issues
#2506

### Verification steps

Select a site and mark its selected soil ID match. Verify that the match is visible on both the dashboard and soil ID screens. Select another match and verify that the displayed match changes. Un-select the match and verify that the top match is displayed instead.
